### ISSUE TITLE
Missing Convlstm related keras layer

### ISSUE THREAD

**Issue by dding3** [intel-analytics/analytics-zoo#724]
Since we are trying to use keras api for seq2seq pr, it would be great if we can have convlstm(2D/3D) keras layers.

**Comment by hkvision** [comment]
We have ConvLSTM2D: https://github.com/intel-analytics/analytics-zoo/blob/master/zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/ConvLSTM2D.scala
I will add ConvLSTM3D soon.
Also feel free to tell me if there's any argument not exposed in Keras API but needed in seq2seq.

**Comment by hkvision** [comment]
Code in ConvLSTM2D is not well written currently. I will refactor the code and add ConvLSTM3D after the fix of Recurrent.

**Comment by dding3** [comment]
Yes, please extend ConvLSTM2D from zoo.keras.Recurrent other than BigDLConvLSTM2D

**Comment by dding3** [comment]
On a second thought, currently ConvLSTM2D cannot be used, maybe we can fix it by extend it from bigdl.keras.Recurrent before zoo.Recurrent is fixed?

**Comment by hkvision** [comment]
OK. Then I will fix it today.

**Comment by hkvision** [comment]
It can be used (i.e. added to a Sequential). The problem is doesn't extend Recurrent and is not treated as a Recurrent layer. Will refactor it afterwards.

**Comment by anpunjabi** [comment]
Any updates on the status of ConvLSTM3D?

**Comment by hkvision** [comment]
Hi, @anpunjabi Thanks for your reminder. We will add ConvLSTM3D very soon.

**Comment by anpunjabi** [comment]
Thank you very much, and let me know when the code is pushed. I have a 3D recurrent architecture that will greatly benefit from this addition.

**Comment by hkvision** [comment]
Hi, @anpunjabi we have added ConvLSTM3D into master, which currently supports 'same' padding and 'channel_first' data format.
See here for Python API: https://github.com/intel-analytics/analytics-zoo/blob/master/pyzoo/zoo/pipeline/api/keras/layers/convolutional_recurrent.py#L92
See here for Scala API: https://github.com/intel-analytics/analytics-zoo/blob/master/zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/ConvLSTM3D.scala
Hope that it can meet your needs.

**Comment by hkvision** [comment]
I will close this issue first. If you have further requests about ConvLSTM layers, feel free to raise another issue :).
